test/extended/router/wildcard.go

```diff
@@ -0,0 +1,93 @@
+package router
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[Conformance][Area:Networking][Feature:Router]", func() {
+	defer g.GinkgoRecover()
+	var (
+		fixture = exutil.FixturePath("testdata", "reencrypt-serving-cert.yaml")
+		oc      *exutil.CLI
+
+		ns string
+	)
+
+	oc = exutil.NewCLI("router-reencrypt", exutil.KubeConfigPath())
+
+	g.BeforeEach(func() {
+		ns = oc.KubeFramework().Namespace.Name
+		// TODO: Only run these tests if the cluster config ingress domain is provided.
+	})
+
+	g.Describe("The default ClusterIngress", func() {
+		g.It("should support default wildcard reencrypt routes through external DNS", func() {
+			execPodName := exutil.CreateExecPodOrFail(oc.AdminKubeClient().Core(), ns, "execpod")
+			defer func() { oc.AdminKubeClient().Core().Pods(ns).Delete(execPodName, metav1.NewDeleteOptions(1)) }()
+
+			g.By(fmt.Sprintf("deploying a service using a reencrypt route using only defaults"))
+			err := oc.Run("create").Args("-f", fixture).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			var hostname string
+			err = wait.Poll(time.Second, changeTimeoutSeconds*time.Second, func() (bool, error) {
+				route, err := oc.RouteClient().Route().Routes(ns).Get("serving-cert", metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				if len(route.Status.Ingress) == 0 || len(route.Status.Ingress[0].Host) == 0 {
+					return false, nil
+				}
+				hostname = route.Status.Ingress[0].Host
+				return true, nil
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			url := "https://" + hostname
+			g.By(fmt.Sprintf("verifying the route serves 200 from %s", url))
+			err = waitForURLOK(ns, execPodName, url, changeTimeoutSeconds)
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+	})
+})
+
+func waitForURLOK(ns, execPodName, url string, timeoutSeconds int) error {
+	cmd := fmt.Sprintf(`
+		set -e
+		for i in $(seq 1 %d); do
+			code=$( curl -k -s -o /dev/null -w '%%{http_code}\n' %q ) || rc=$?
```

Comment:
I'd suggest adding `-m 1` to the `curl` command to time out after 1 second.  If you want a higher timeout value, or if you want to observe `timeoutSeconds` more strictly, you could add `if [[ $SECONDS -gt %d ]]; then echo timed out; exit 1; fi` to the loop.

Reply:
Done